src/charm.py

```diff
@@ -52,18 +52,39 @@ def __init__(self, *args: Any) -> None:
             host=f"{self.app.name}-endpoints.{self.model.name}.svc.cluster.local",
             strip_prefix=True,
         )
+        self.framework.observe(self.on.config_changed, self._on_config_changed)
+        self.framework.observe(self.on.reset_instance_action, self._reset_instance_action)
 
     def _on_config_changed(self, event: ops.HookEvent) -> None:
         """Handle changed configuration.
 
         Args:
             event: Event triggering after config is changed.
         """
+        try:
+            self._charm_state = CharmState.from_charm(charm=self)
+        except CharmConfigInvalidError as exc:
+            self.model.unit.status = ops.BlockedStatus(exc.msg)
+            return
         container = self.unit.get_container(SYNAPSE_CONTAINER_NAME)
         if not container.can_connect():
             event.defer()
             self.unit.status = ops.WaitingStatus("Waiting for pebble")
             return
+        server_name_configured = self._synapse.server_name_configured(container)
```

Comment:
perhaps `configured_server_name`? At the moment it reads like a boolean value of whether the server name is configured, I'm guessing this is either the configured server name or None?

Reply:
Thanks for the suggestion! Changed.